src/test/java/org/junit/tests/assertion/AssertionTest.java

```diff
@@ -674,4 +677,80 @@ public void assertNotEqualsIgnoresDeltaOnNaN() {
     public void assertNotEqualsIgnoresFloatDeltaOnNaN() {
         assertNotEquals(Float.NaN, Float.NaN, 1f);
     }
+
+    @Test(expected = AssertionError.class)
+    public void expectThrowsRequiresAnExceptionToBeThrown() {
+        expectThrows(nonThrowingRunnable());
+    }
+
+    @Test
+    public void expectThrowsIncludesNoMessageByDefault() {
+        try {
+            expectThrows(nonThrowingRunnable());
+            fail();
+        } catch (AssertionError ex) {
+            assertNull(ex.getMessage());
+        }
+    }
+
+    @Test
+    public void expectThrowsIncludesTheSuppliedMessage() {
+        try {
+            expectThrows("message", nonThrowingRunnable());
+            fail();
+        } catch (AssertionError ex) {
+            assertEquals("message", ex.getMessage());
+        }
+    }
+
+    @Test
+    public void expectThrowsReturnsTheSameObjectThrown() {
+        NullPointerException npe = new NullPointerException();
+
+        Throwable throwable = expectThrows(throwingRunnable(npe));
+
+        assertSame(npe, throwable);
+    }
+
+    @Test(expected = ClassCastException.class)
```

Comment:
Now that I think about it, we should be more concerned about losing the exception that actually _was_ caught. That's the main piece of information you would need to debug a problem in the underlying SUT. (Note that fixing this is orthogonal to whether the expected exception type is specified/required.)